drivers/SmartThings/samsung-audio/src/handlers.lua

```diff
@@ -106,7 +106,10 @@ end
 
 function CapabilityHandlers.handle_set_volume(driver, device, cmd)
   local ip = device:get_field("ip")
-  command.set_volume(ip, cmd.args.volume)
+  local vol = command.set_volume(ip, cmd.args.volume)
+  if vol then
+    device:emit_event(capabilities.audioVolume.volume(tonumber(vol.volume)))
+  end
```

Comment:
Could this same pattern be used with other commands? Emitting the event if we get a response?

Reply:
> Could this same pattern be used with other commands? Emitting the event if we get a response?

Done